JavaScript/Default.sublime-keymap

```diff
@@ -0,0 +1,54 @@
+[
+    // Auto-pair backticks
+    { "keys": ["`"], "command": "insert_snippet", "args": {"contents": "`$0`"}, "context":
+        [
+            { "key": "selector", "operator": "equal", "operand": "source.js - string" },
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selection_empty", "operator": "equal", "operand": true, "match_all": true },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^(?:\t| |\\)|]|\\}|>|$)", "match_all": true },
+        ]
+    },
+    { "keys": ["`"], "command": "insert_snippet", "args": {"contents": "`${0:$SELECTION}`"}, "context":
+        [
+            { "key": "selector", "operator": "equal", "operand": "source.js - string" },
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selection_empty", "operator": "equal", "operand": false, "match_all": true }
+        ]
+    },
+    { "keys": ["`"], "command": "move", "args": {"by": "characters", "forward": true}, "context":
+        [
+            { "key": "selector", "operator": "equal", "operand": "source.js" },
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selection_empty", "operator": "equal", "operand": true, "match_all": true },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^`", "match_all": true },
+            { "key": "selector", "operator": "not_equal", "operand": "punctuation.definition.string.begin", "match_all": true },
+        ]
+    },
+    { "keys": ["backspace"], "command": "run_macro_file", "args": {"file": "res://Packages/Default/Delete Left Right.sublime-macro"}, "context":
+        [
+            { "key": "selector", "operator": "equal", "operand": "source.js string.template" },
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selection_empty", "operator": "equal", "operand": true, "match_all": true },
+            { "key": "preceding_text", "operator": "regex_contains", "operand": "`$", "match_all": true },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^`", "match_all": true },
+            { "key": "selector", "operator": "not_equal", "operand": "punctuation.definition.string.begin", "match_all": true },
+        ]
+    },
+
+    // Auto-pair interpolation
+    { "keys": ["{"], "command": "insert_snippet", "args": {"contents": "{$0}"}, "context":
```

Comment:
The Default keymap has extra conditions. It won't insert a brace here if you type `` ` ``, `$`, `{`. This binding doesn't need those conditions, so it will work in that situation.

Reply:
Hmm.

1. Including this commit typing ``` ` ``` and `$` would add ``` `${}` ``` by the other two auto-pair rules.
2. Typing all three on a vanilla ST setup adds ``` `${} ``` with curly-braces being auto-paired.

But your are right with not working default auto-pairing if one tries to insert an embedded script into existing backticks ``` `` ```. The `following_text` wouldn't match then. Didn't catch it on a first glance as I use already patched key bindings for auto-pairing as the default rules are not ideal and cause auto-pairing to fail in some situations.

Reply:
I think that in a perfect world we should find a better solution than the existing lookaheads for the Default pairing rules. It's tricky, though, because those rules are designed to work for plain text. Barring a change there, I think that this override is necessary to provide a good user experience in JavaScript.

Reply:
I finally agree. It avoids some possible troubles when using auto-pairing in templates.